### PR TITLE
Eh 897 refactoring

### PR DESCRIPTION
**Pull request by vesaruu**
Started to refactor Details component as it has grown quite large with it's expand/collapse functionality. Extracted two parts of expanded details to components. This work will continue but like to integrate continuously.
Renamed LearningPeriods to OsaamisenHankkimistavat and Demonstrations to OsaamisenOsoittamiset.

### CODE REVIEW

File: shared/components/StudyInfo/Details.tsx [hunk 1]
```diff
@@ -74,53 +77,102 @@ const VerificationTitleExpanded = styled("strong")`
...
+}) => (
+  <ShareDialog
+    active={hasActiveShare && shareType === "tyossaoppiminen"}
+    background={fadedColor}
+    koodiUri={koodiUri || ""}
+    type="tyossaoppiminen"
+    instructor={instructor}
+    defaultPeriod={defaultPeriod}
+  >
+    {osaamisenHankkimistavat.map((period, i) => {
```
Comment (J-Kallunki) [inline shared/components/StudyInfo/Details.tsx:130]: period = osaamisenHankkimistapa ?

File: shared/components/StudyInfo/Details.tsx [hunk 2]
```diff
@@ -74,53 +77,102 @@ const VerificationTitleExpanded = styled("strong")`
...
+}: {
+  hasActiveShare: boolean
   toggle: (name: ToggleableItems) => () => void
-  verificationProcess?: TodentamisenProsessi
-  koulutuksenJarjestaja?: IOrganisaatio
-}
+  intl: InjectedIntl
+}) => (
+  <>
+    {!hasActiveShare && (
```
[the remark this answers is not in the document]
Comment (vesaruu) [reply shared/components/StudyInfo/Details.tsx:90]: Yes true, negation of boolean is confusing but wanted to just extract the current implementation as is and not change any functionality. I try to resist the temptation to fix things when refactoring as it's very easy to lose focus and sidetrack too much.

File: shared/components/StudyInfo/Details.tsx [hunk 3]
```diff
@@ -74,53 +77,102 @@ const VerificationTitleExpanded = styled("strong")`
...
+  shareType,
+  fadedColor,
+  koodiUri,
+  instructor,
+  defaultPeriod,
+  osaamisenHankkimistavat
+}: {
+  hasActiveShare: boolean
+  shareType?: ShareType | ""
+  fadedColor: string
```
[the remark this answers is not in the document]
Comment (vesaruu) [reply shared/components/StudyInfo/Details.tsx:116]: Same as above, extracted the current implementation as is.

File: shared/components/StudyInfo/Details.tsx (hunk 1 above)
Comment (vesaruu) [reply shared/components/StudyInfo/Details.tsx]: True, didn't spot this one, I'll change

**Pull request merged**